cogs/customreactions.py

```diff
@@ -0,0 +1,1255 @@
+                if all(
+                    (not msg_user
+                     or msg.author == msg_user, msg.channel == ctx.channel)):
+                    if msg.attachments:
+                        # in python 3.7, rewrite as
+                        # asyncio.create_task(ctx.send([...]))
+                        # (the get_event_loop() part isn't necessary)
+                        loop = asyncio.get_event_loop()
+                        loop.create_task(
+                            ctx.send("Attachments cannot be used, "
+                                     "but you may use URLs"))
+                    else:
+                        return True
+
+            return msg_check
+
+        def get_number_check(msg_user=None, number_range=None):
+            def number_check(msg):
+                if msg.content.isdigit():
+                    return all((not msg_user
+                                or msg.author == msg_user, not number_range
+                                or int(msg.content) in number_range,
+                                msg.channel == ctx.channel))
+
+            return number_check
+
+        async def wait_for_reaction(message):
+            try:
+                reaction, user = await self.bot.wait_for(
+                    'reaction_add',
+                    check=get_reaction_check(reaction_user=main_user),
+                    timeout=60)
+            except asyncio.TimeoutError:
+                await message.clear_reactions()
+                await message.edit(
+                    embed=discord.Embed(title=CUSTOM_REACTION_TIMEOUT),
+                    delete_after=60)
+                return
+            return reaction, user
+
+        async def wait_for_message(message):
+            try:
+                msg = await self.bot.wait_for(
+                    'message',
+                    check=get_msg_check(msg_user=main_user),
+                    timeout=60)
+            except asyncio.TimeoutError:
+                await message.clear_reactions()
+                await message.edit(
+                    embed=discord.Embed(title=CUSTOM_REACTION_TIMEOUT),
+                    delete_after=60)
+                return
+            content = msg.content
+            await msg.delete()
+            return content
+
+        async def add_multiple_reactions(message, reactions):
+            for reaction in reactions:
+                await message.add_reaction(reaction)
+
+        async def add_yes_or_no_reactions(message):
+            await add_multiple_reactions(
+                message, (EMOJI['zero'], EMOJI['one'], EMOJI['stop_button']))
+
+        async def add_control_reactions(message):
+            await add_multiple_reactions(
+                message, (EMOJI['rewind'], EMOJI['arrow_backward'],
+                          EMOJI['arrow_forward'], EMOJI['fast_forward'],
+                          EMOJI['stop_button'], EMOJI['ok']))
+
+        async def create_assistant(message, is_moderator):
```

Comment:
in the future, avoid overriding stuff like this (is_moderator is defined outside this scope too, which would make it confusing if it was mutated)

Reply:
Resolved in 5418e37